test/parallel/test-tls-psk-circuit.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+const common = require('../common');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const tls = require('tls');
+
+const CIPHERS = 'PSK+HIGH';
+const USERS = {
+  UserA: 'd731ef57be09e5204f0b205b60627028',
+  UserB: '82072606b502b0f4025e90eb75fe137d'
+};
+const CLIENT_IDENTITIES = [
+  { psk: USERS.UserA, identity: 'UserA' },
+  { psk: USERS.UserB, identity: 'UserB' },
+  // unrecognized user should fail handshake
+  { psk: USERS.UserB, identity: 'UserC' },
+  // recognized user but incorrect secret should fail handshake
+  { psk: USERS.UserA, identity: 'UserB' },
+  { psk: USERS.UserB, identity: 'UserB' }
```

Comment:
Is there a specific reason why you are testing `UserB` / `UserB` again? Did you mean to use `identity: 'UserA'`?

Reply:
@tniessen This is just a test to ensure that even after a failed authentication, that the valid user is still able to authenticate.

Reply:
Thanks for the clarification, great!